include/seastar/http/request.hh

```diff
@@ -212,6 +212,26 @@ struct request {
      */
     void write_body(const sstring& content_type, noncopyable_function<future<>(output_stream<char>&&)>&& body_writer);
 
+    /**
+     * \brief Use an output stream to write the message body
+     *
+     * When a handler needs to use an output stream it should call this method
+     * with a function.
```

Comment:
Please something about this overload being specifically about the case that the output's length is known in advance, but the full output is not available as a single buffer. If the output length is not known in advance, use <some other overload>, and if the full body is available in a single buffer use <some other overload>.